src/cargo/core/compiler/mod.rs

```diff
@@ -275,6 +276,18 @@ fn rustc<'a, 'cfg>(
             }
         }
 
+        fn internal_if_simple_exit_code(err: Error) -> Error {
+            if err
+                .downcast_ref::<ProcessError>()
+                .as_ref()
+                .and_then(|perr| perr.exit.and_then(|e| e.code()))
```

Comment:
I think rustc only has one erroneous exit code itself, so could we perhaps whitelist just that one exit code to handle cases like Windows to ensure that Windows also prints segfaults by default?

Reply:
If you think it's safe. It looks like rustdoc uses the same code, I'm just thinking of wrappers using different codes.  But maybe that's a good thing?

Reply:
An alternative is to filter out any exit code less than 128. On Unix I think that's the maximal error code and any portable program won't be using upper codes on Windows. (and on Windows all the abnormal exit codes are far above 128). 

Do you think we should perhaps implement that sort of filtering to be a bit more conservative?

Reply:
Sounds good, also means I don't need to fix the 1.28 failure.